src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -2079,17 +2079,18 @@ void EIRPlantLoopHeatPump::isPlantInletOrOutlet(EnergyPlusData &state)
         for (auto thisLoopSide : thisPlant.LoopSide) {
             if (this->loadSideNodes.inlet == thisLoopSide.NodeNumIn) {
                 loadSideIsPlantInlet = true;
-                break;
             }
             if (this->sourceSideNodes.outlet == thisLoopSide.NodeNumOut) {
                 sourceSideIsPlantOutlet = true;
-                break;
```

Comment:
If the order of the PlantLoop objects is such that either of these checks is last in the loops, the break will bypass the conditional check below that identifies the WWHP as a heat recovery unit. Note the order of the PlantLoop objects is different than the original example file and was purposely changed to show this defect.

Reply:
This fix makes sense.  Thanks @rraustad